### ISSUE TITLE
Build and release ruleset_converter

### ISSUE THREAD

**Issue by uazo** [uazo/bromite-buildtools#256]
as suggested in https://github.com/uazo/bromite-buildtools/issues/198#issuecomment-1598435955

**Comment by torvic9** [comment]
I built a new one including the `subresource_indexing_tool` and used both tools to create a new binary filter list from a few adblock sources (EasyList, Fanboy, Hagezi). 

However it seems that the newly generated filters.dat still doesn't do any ad blocking.

---
Commands used:
```
./ruleset_converter --input_files=xxx,yyy,zzz --input_format=filter-list --output_file=filters-unindexed.dat --output_format=unindexed-ruleset
./subresource_indexing_tools filters-unindexed.dat filters.dat
```

**Comment by uazo** [comment]
> newly generated filters.dat

please give me the url to check it out

**Comment by torvic9** [comment]
Here you are: https://gitlab.com/torvic9/adfilter/-/blob/master/filters.dat

(this is just a temporary file and will get removed soon-ish)

**Comment by uazo** [comment]
> you probably need the [raw link](https://gitlab.com/torvic9/adfilter/-/raw/master/filters.dat)

yes, I saw, that was the error:

```
DownloadFiltersTask: fetching URL 'https://gitlab.com/torvic9/adfilter/-/raw/master/filters.dat?inline=false' 
with method GET (no Last-Modified header)
DownloadFiltersTask: downloaded 0 bytes in 2237ms from 'https://gitlab.com/torvic9/adfilter/-/raw/master/filters.dat?inline=false' to '' 
with net_error -12 and error 200
```

net_error 12 is `INSUFFICIENT_RESOURCES` see https://source.chromium.org/chromium/chromium/src/+/main:net/base/net_error_list.h

The file is probably too large (14 mb).
It was a limitation that had already come out in another issue in bromite.
let's see if I can increase it.

**Comment by uazo** [comment]
> let's see if I can increase it.

yes, the current limit is 10 mb, I can easily increase it to 50.
perhaps we have found the bug reported in another issue...

but I probably have to add a button to force the update, because otherwise the last-modified-value in the header of the old downloaded file is checked and the new download does not take place.

**Comment by torvic9** [comment]
Ah good catch!
I will try with a smaller <10M file in the meantime.

Indexed filters are more than twice as big as the unindexed ones.

> in debug I have already done it, file downloaded and correctly parsed. can you tell me what you would use to verify that it works?

Hmm I disable the alternative list based adblocking and then just open a website which is known to have ads that are blocked by a correct filter list.

For example https://welt.de shouldn't show (many) ads when the EasyList-Germany filter is used and active.

**Comment by uazo** [comment]
OK, I thought you were using some particular tool.

> For example https://welt.de/ shouldn't show (many) ads when the EasyList-Germany filter is used and active.

well, I don't know German but I don't see any advertising!

I don't actually see them even if I only use the other (eyeo) adblock..
question, why don't you use that (eyeo)?

**Comment by torvic9** [comment]
You must disable all ad blocking to see them. There are not many but they are clearly visible.
With Eyeo adblock + EasyListGerman they are (mostly) gone.

ABP is (IMO) not complete enough but of course much better than nothing at all.

**Comment by uazo** [comment]
Sorry, perhaps I have explained myself wrongly.
i meant, instead of using ruleset_converted, to point directly to the source of the rules in adp (https://gitlab.com/torvic9/adfilter/-/raw/master/filters.txt?inline=false) which, however, must be written with the adp syntax

**Comment by torvic9** [comment]
I tried adding that "filters.txt" (which should be in ABP format) to the new Eyeo based ad blocking but that doesn't work (adding an existing list like Hagezi's doesn't work either).

This txt file was indeed created with ruleset_converter in order to combine several different lists into one.

**Comment by uazo** [comment]
you're right, it crashes

```
  0000000007600f9e  adblock::SubscriptionDownloaderImpl::OnDownloadFinished(GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>)  ../../components/adblock/core/subscription/subscription_downloader_impl.cc:179:3
  0000000007601e80  void base::internal::FunctorTraits<void (adblock::SubscriptionDownloaderImpl::*)(GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>), void>::Invoke<void (adblock::SubscriptionDownloaderImpl::*)(GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>), base::WeakPtr<adblock::SubscriptionDownloaderImpl> const&, GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>>(void (adblock::SubscriptionDownloaderImpl::*)(GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>), base::WeakPtr<adblock::SubscriptionDownloaderImpl> const&, GURL const&, base::FilePath&&, scoped_refptr<net::HttpResponseHeaders>&&)  ../../base/functional/bind_internal.h:746:12
  00000000075fa02d  base::RepeatingCallback<void (GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>)>::Run(GURL const&, base::FilePath, scoped_refptr<net::HttpResponseHeaders>) const &  ../../base/functional/callback.h:333:12
  00000000075f9e2e  adblock::OngoingSubscriptionRequestImpl::OnDownloadFinished(base::FilePath)       ../../components/adblock/core/subscription/ongoing_subscription_request_impl.cc:175:22
  00000000075fa4ec  void base::internal::FunctorTraits<void (adblock::OngoingSubscriptionRequestImpl::*)(base::FilePath), void>::Invoke<void (adblock::OngoingSubscriptionRequestImpl::*)(base::FilePath), adblock::OngoingSubscriptionRequestImpl*, base::FilePath>(void (adblock::OngoingSubscriptionRequestImpl::*)(base::FilePath), adblock::OngoingSubscriptionRequestImpl*&&, base::FilePath&&)  ../../base/functional/bind_internal.h:746:12
  00000000075fa48a  void base::internal::InvokeHelper<false, void, 0ul>::MakeItSo<void (adblock::OngoingSubscriptionRequestImpl::*)(base::FilePath), std::Cr::tuple<base::internal::UnretainedWrapper<adblock::OngoingSubscriptionRequestImpl, base::unretained_traits::MayNotDangle, (base::RawPtrTraits)0>>, base::FilePath>(void (adblock::OngoingSubscriptionRequestImpl::*&&)(base::FilePath), std::Cr::tuple<base::internal::UnretainedWrapper<adblock::OngoingSubscriptionRequestImpl, base::unretained_traits::MayNotDangle, (base::RawPtrTraits)0>>&&, base::FilePath&&)  ../../base/functional/bind_internal.h:925:12
  000000000b73803f  base::OnceCallback<void (base::FilePath)>::Run(base::FilePath) &&                 ../../base/functional/callback.h:152:12
  000000000b736647  network::(anonymous namespace)::SaveToFileBodyHandler::NotifyConsumerOfCompletion(bool)  ../../services/network/public/cpp/simple_url_loader.cc:886:52
  000000000b734005  network::(anonymous namespace)::SimpleURLLoaderImpl::FinishWithResult(int)        ../../services/network/public/cpp/simple_url_loader.cc:1631:18
  000000000b73415f  network::(anonymous namespace)::SimpleURLLoaderImpl::MaybeComplete()              ../../services/network/public/cpp/simple_url_loader.cc:1939:3
  000000000b733e56  network::(anonymous namespace)::SimpleURLLoaderImpl::OnBodyHandlerDone(net::Error, long)  ../../services/network/public/cpp/simple_url_loader.cc:1580:3

```

**Comment by uazo** [comment]
> Hmm interesting, perhaps also too big?

no, looking at the code it seems that I probably activated a feature that is still in development, I will try to fix it.

> https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/light.txt

`[ERROR:metadata.cc(43)] [eyeo] Invalid filter list. Should start with [Adblock Plus <x>.<y>].`

**Comment by torvic9** [comment]
> `[ERROR:metadata.cc(43)] [eyeo] Invalid filter list. Should start with [Adblock Plus <x>.<y>].`

Weird error message, as my list doesn't start with that either?
Anyway should be easy to fix.

**Comment by uazo** [comment]
> no, looking at the code it seems that I probably activated a feature that is still in development, I will try to fix it.

I found the error to be related to the presence of the parameters in the url (the "?inline=false")
I will fix in the next version

> EDIT: added the tag to my list for good measure.

what the code does is a simple formal check.
Removing the check, the file to load correctly. I will probably remove it.

**Comment by uazo** [comment]
- removed the need of header in rules
- allowed the use of urls with parameters in subscriptions

fixed in https://github.com/uazo/bromite/commit/a0cfe08ebdfae5b1d88912dcfc121649a97fc76d

**Comment by uazo** [comment]
Since pre-built tools are also available here https://github.com/xarantolus/subresource_filter_tools/releases I'm closing with issue.

@torvic9 please check the new version and in case of bug open a new issue.
